### PR TITLE
updated postgres column varchar lengths to be 8703

### PR DESCRIPTION
**Pull request by jcolson**
http://sprunge.us/RHHPXX

`2019-02-11 10:26:06 [Warn ] Migration Failed!
2019-02-11 10:26:06 [Warn ] AbstractSqlMigrationReader::transferMo(): unable to transfer Migratable Object of type IrcServer!
2019-02-11 10:26:06 [Warn ] WriterError:
2019-02-11 10:26:06 [Warn ]   executed Query:
2019-02-11 10:26:06 [Warn ] INSERT INTO ircserver (serverid, userid, networkid, hostname, port, password, ssl, sslversion, useproxy, proxytype, proxyhost, proxyport, proxyuser, proxypass, sslverify)
VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
2019-02-11 10:26:06 [Warn ]   bound Values:
2019-02-11 10:26:06 [Warn ] 0 :  10
2019-02-11 10:26:06 [Warn ] 1 :  1
2019-02-11 10:26:06 [Warn ] 2 :  2
2019-02-11 10:26:06 [Warn ] 3 :  localhost
2019-02-11 10:26:06 [Warn ] 4 :  6696
2019-02-11 10:26:06 [Warn ] 5 :  xoxp-xxxxxxxxxx-xxxxxxxxxxx-xxxxxxxxxxxx-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

2019-02-11 10:26:06 [Warn ] 6 :  true
2019-02-11 10:26:06 [Warn ] 7 :  0
2019-02-11 10:26:06 [Warn ] 8 :  false
2019-02-11 10:26:06 [Warn ] 9 :  1
2019-02-11 10:26:06 [Warn ] 10 :  localhost
2019-02-11 10:26:06 [Warn ] 11 :  8080
2019-02-11 10:26:06 [Warn ] 12 :  
2019-02-11 10:26:06 [Warn ] 13 :  
2019-02-11 10:26:06 [Warn ] 14 :  false
2019-02-11 10:26:06 [Warn ]   Error Number: 22001
2019-02-11 10:26:06 [Warn ]   Error Message: "ERROR:  value too long for type character varying(64)\n(22001) QPSQL: Unable to create query"
2019-02-11 10:26:06 [Warn ] Unable to migrate storage backend! (No migration writer for PostgreSQL)`

### CODE REVIEW

**Comment by digitalcircuit** [comment]
Given we have other places with `VARCHAR(64)`, should we consider bumping it everywhere?

There's some discussion on this over on StackOverflow; supposedly, there's not a noticeable performance impact: https://stackoverflow.com/questions/1067061/does-a-varchar-fields-declared-size-have-any-impact-in-postgresql

I'd vote for increasing in other places if we increase here, since if it's allowed on SQLite Quassel itself already handles it fine.

Regardless of chosen lengths, this pull request should apply a PostgreSQL migration so existing databases benefit from this change and are consistent with a newly-created database.  It should be as straightforward as making a new folder inside [`src/core/SQL/PostgreSQL/version`](https://github.com/quassel/quassel/tree/master/src/core/SQL/PostgreSQL/version ) and adding the necessary queries to alter the database columns without losing the existing data.

**Comment by mamarley** [comment]
I have no objection to merging this as-is, but regarding @digitalcircuit 's comment, I think it would be a good idea to limit the field length just to prevent DoS and other potential issues from having overlength strings.  However, I think that should ideally be done in the code rather than in the database because right now, if you try to create a network with an overlength name, it will fail and hang while syncing the data to the core.  It would be more user-friendly to display a proper error message.

**Comment by justjanne** [comment]
I’d propose just setting it to VARCHAR(512), as IRC messages are limited in size to 512 bytes anyway.

**Comment by jcolson** [comment]
so 512?  any other fieds i should alter as well (proxy username and password were mentioned)?

**Comment by digitalcircuit** [comment]
Unless I'm misunderstanding, it should be fine to adjust all the `VARCHAR(#)` in `quassel/src/core/SQL/PostgreSQL/setup*.sql` (all the setup files) to be `VARCHAR(512)`.  Then, also add the schema upgrade queries for all of them.

Later, the client could add appropriate warnings for text being too long.

(*Maybe even 2048 since the bad situation is having the client hang?*)

**Comment by jcolson** [comment]
another column with an issue ... https://bugs.quassel-irc.org/issues/1526

so 2048 across the board? if so I'll modify PR

**Comment by justjanne** [comment]
The maximum length of an IRCv3 message would be 8191 chars for tags plus 512 chars for content. So we might end up worst case with 8703, which should always be enough.

**Comment by digitalcircuit** [comment]
@Sput42 I think the `needs more work` label can be removed.  I only noticed [one minor additional cleanup that could be done](https://github.com/quassel/quassel/pull/478#discussion_r296358454 ), and functionality all seems intact.

File: src/core/SQL/PostgreSQL/setup_000_quasseluser.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 CREATE TABLE quasseluser (
        userid serial NOT NULL PRIMARY KEY,
-       username varchar(64) UNIQUE NOT NULL,
+       username varchar(8703) UNIQUE NOT NULL,
        password TEXT NOT NULL,
        hashversion integer NOT NULL DEFAULT 0,
-       authenticator varchar(64) NOT NULL DEFAULT 'Database'
+       authenticator varchar(8703) NOT NULL DEFAULT 'Database'
```
[the remark this answers is not in the document]
Comment (jcolson) [reply src/core/SQL/PostgreSQL/setup_000_quasseluser.sql]: done here --> 7fc7701

**Comment by jcolson** [comment]
the continuous-integration/appveyor/pr issue doesn't look related to this change?

[131/371] Building CXX object src\client\CMakeFiles\quassel_client.dir\buffermodel.cpp.obj
FAILED: src/client/CMakeFiles/quassel_client.dir/buffermodel.cpp.obj

**Comment by justjanne** [comment]
@jcolson that’s something else, just rebase the PR based on the latest master

**Comment by digitalcircuit** [comment]
In the process of expanding IRCv3 capability handling (to allow disabling specific IRCv3 capabilities), I'm needing to introduce another database schema change, so to simplify avoiding conflicts, I've combined these commits into a new one, setting `jcolson` as the commit author:
https://github.com/quassel/quassel/compare/master...digitalcircuit:ft-ircv3-specced-out

@jcolson , let me know if you'd like any changes!

@Sput42 When reviewing PRs, if my IRCv3 PR isn't yet created (*I'll edit this comment when it's made*), feel free to merge this PR instead, and I'll just rebase atop master.

**Comment by jcolson** [comment]
@digitalcircuit 8703 is good with me!

**Comment by digitalcircuit** [comment]
@jcolson Ack, pardon the confusion.  I meant to ask if the way I've [given you attribution in pull request #545](https://github.com/quassel/quassel/pull/545/commits ) is fine by you.

I'm also fine with letting Sput review and merge these pull requests separately, it might just make things simpler to bundle them (since I have to make a database schema change as well).

**Comment by jcolson** [comment]
@digitalcircuit you bet, totally fine by me.

**Comment by digitalcircuit** [comment]
Small reminder/heads up for @Sput42:

This pull request has now been fully incorporated/subsumed into ["common: IRCv3++, more IRCv3, allow disabling IRCv3 caps #545"](https://github.com/quassel/quassel/pull/545 ) - I'd request that you **do not** merge PR #478 unless I'm unable to finish the IRCv3 testing in time.

**Pull request closed without merging**